components/app-framework/src/widgets/log-viewer/log-viewer.directive.js

```diff
@@ -396,17 +401,33 @@
 
       // Handle streaming logs
       function requestStreamingLog() {
+        function onOpen(event) {
+          $log.debug('WebSocket connection opened', event);
+          logViewer.streaming = STREAMING_STATUS.ONLINE;
+          safeApply();
+        }
+
         resetLog();
 
-        var websocketUrl = logViewer.websocketUrl;
-        if (_.isUndefined(websocketUrl)) {
+        if (logViewer.websocketUrl) {
+          logViewer.streaming = STREAMING_STATUS.CONNECTING;
+          logViewer.webSocketConnection = $websocket(logViewer.websocketUrl, null, {
+            reconnectIfNotNormalClose: false
+          });
+        } else if (logViewer.websocket) {
+          if (logViewer.webSocketConnection) {
+            closeWebSocket();//logViewer.webSocketConnection.close(true)
+          }
+          logViewer.webSocketConnection = logViewer.websocket;
+          if (logViewer.webSocketConnection.readyState === 1) {
```

Comment:
We might miss the case where we're opening a connection but haven't completed before the synchronous code further down is hit though?